File: Day 3 - Sorting Techniques/01_SelectionSort.js
function selectionSort(nums) {
    
    for (let i = 0; i < nums.length - 1; i++) {
        let min_idx  = i;
        for (let j = i+1; j < nums.length; j++) {
            if (nums[j] < nums[min_idx]) {
                min_idx = j;
            }
        }
        if (i != min_idx) {
            [nums[i], nums[min_idx]] = [nums[min_idx], nums[i]];
        }
    }
    return nums;
}


console.log(selectionSort([112, -22, 0, 11, 12, 100, 23, -1]));

File: 03_Sorting Techniques/01_SelectionSort/01_SelectionSort.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;


void selectionSort(vector<int> &nums) {

    for (int i = 0; i < nums.size() - 1; i++) {
        int min_idx = i;
        for (int j = i + 1; j < nums.size(); j++) {
            if (nums[min_idx] > nums[j]) {
                min_idx = j;
            }
        }
        if (min_idx != i) swap(nums[i], nums[min_idx]);
    }
}

int main() {
    vector<int> arg = {100, 2, 0, -1, -23, -100};
    selectionSort(arg);
    for (auto x: arg) {
        cout << x << ' ';
    }
    cout << endl;
    return 0;
}